spec/components/concern/has_custom_fields_spec.rb

```diff
@@ -187,6 +187,55 @@ class CustomFieldsTestItemCustomField < ActiveRecord::Base
       expect(test_item2.custom_fields).to eq("sixto" => "rodriguez", "de" => "la playa")
     end
 
+    it "supports arrays in json fields" do
+      field_type = "json_array"
+      CustomFieldsTestItem.register_custom_field_type(field_type, :json)
+
+      item = CustomFieldsTestItem.new
+      item.custom_fields = {
+        "json_array" => [{ a: "test" }, { b: "another" }]
+      }
+      item.save
+
+      item = CustomFieldsTestItem.find(item.id)
+
+      expect(item.custom_fields[field_type]).to eq(
+          [{ "a" => "test" }, { "b" => "another" }]
+      )
+
+      item.custom_fields["json_array"] = ['a']
+      item.save
+
+      item = CustomFieldsTestItem.find(item.id)
+
+      expect(item.custom_fields[field_type]).to eq(["a"])
+
+      item.custom_fields["json_array"] = ['a', 'b']
```

Comment:
I'm wondering if we need this test case here since the number of items in the array isn't testing anything new.

Reply:
I've removed the single item array test case, and kept this instead.